public/js/app.js

```diff
@@ -0,0 +1,110 @@
+// Call function getNextMove(player, coord)
+// player = 'computer' or 'human'
+// coord is an element in [0,1,2,3,4,5,6,7,8]
+// The function returns an element in [0,1,2,3,4,5,6,7,8]
+// as the computer's next move, 'computer', 'human', or 'tie'
+
+computerMoves = []
+humanMoves = []
+board = []
+wins = [[0,1,2],[3,4,5],[6,7,8],[0,3,6],[1,4,7],[2,5,8],[0,4,8],[2,4,6]]
+
+function getNextMove(player, coord){
+    _updateBoard(player, coord);
+    var nextMove = _nextMove(_otherPlayer(player))[0];
+    _updateBoard(_otherPlayer(player), nextMove);
+    console.log(nextMove);
+}
+
+function _updateBoard(player, coord) {
+    board.push(coord);
+    if (player == 'computer')
+        computerMoves.push(coord);
+    else
+        humanMoves.push(coord);
+}
+
+function _undoBoard(player, coord) {
+    board.pop();
+    if (player == 'computer')
+        computerMoves.pop();
+    else
+        humanMoves.pop();
+}
+
+function _otherPlayer(player) {
+    if (player == 'computer')
+        return 'human';
+    else
+        return 'computer';
+}
+
+function _isSubArray (subArray, Array) {
+    for(var i = 0; i < subArray.length; i++) {
+        if($.inArray(subArray[i], Array) == -1)
+            return false;
+    }
+    return true;
+}
+
+function _boardStatus() {
+    for(var i = 0; i < wins.length; i++) {
+        if (_isSubArray(wins[i], computerMoves))
+            return 'computer';
+        else if (_isSubArray(wins[i], humanMoves))
+            return 'human';
+    }
+    if (board.length == 9)
+        return 'tie';
+    else
+        return 'game is still on'
+}
+
+function _getAvailableMoves() {
+    return $.grep([0,1,2,3,4,5,6,7,8], function(n,i) {
+        return $.inArray(n, board) == -1});;
+}
+
+// _nextMove() utilizes the minimax algorithm
+function _nextMove(player) {
+    if(_boardStatus() == 'computer')
+        return ['computer', 1];
+    else if (_boardStatus() == 'human')
+        return ['human', -1];
+    else if (_boardStatus() == 'tie')
+        return ['tie', 0];
+    else {
+        var best = null;
+        var nextMove = null;
+        var availableMoves = _getAvailableMoves();
+
+        for(var i = 0; i < availableMoves.length; i++) {
+            _updateBoard(player, availableMoves[i]);
+            result = _nextMove(_otherPlayer(player));
+            var score = result[1];
+            _undoBoard(player);
+
+            if(best == null || (player == 'computer' && score > best) ||
+              (player == 'human' && score < best)) {
+                best = score;
+                nextMove = availableMoves[i];
+            }
+        }
+        return [nextMove, best];
+    }
+}
+
+////
```

Comment:
fyi... you wanna split error code and message so they aren't tied together.  Now you can alter your message and I don't have to care cause the UI looks for value 1 instead of hard coded message.